SingularityService/src/main/java/com/hubspot/singularity/data/ManagerCache.java

```diff
@@ -0,0 +1,52 @@
+package com.hubspot.singularity.data;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.google.inject.Inject;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ManagerCache<K, V> {
```

Comment:
Check out the ZkCache class. It's basically the same thing as this and might be better suited (at least for the deployManager part of things)

Reply:
The ZkCache just uses a Guava cache and we were hoping to get benefits from the LoadingCache created by CaffeineCache for debouncing requests because we were also getting timeouts at the DeployManager's request, do you still think that I should update the DeployManager's usage?

Reply:
Ah ok, thought the Zk one was more similar. Can keep the separate one then